client/src/ui/molecules/document-survey/index.tsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import { Doc } from "../../../document/types";
+import { ReactComponent as CloseIcon } from "@mdn/dinocons/general/close.svg";
+import "./index.scss";
+import { Survey, SURVEYS } from "./surveys";
+import { getSurveyState, writeSurveyState } from "./util";
+import { useIsServer } from "../../../hooks";
+
+export function DocumentSurvey({ doc }: { doc: Doc }) {
+  const isServer = useIsServer();
+
+  const survey = React.useMemo(
+    () =>
+      SURVEYS.find((survey) => {
+        if (isServer) {
+          return false;
+        }
+
+        if (!survey.show(doc)) {
+          return false;
+        }
+
+        const state = getSurveyState(survey.key);
+
+        return state.random < survey.rate;
+      }),
+    [doc, isServer]
+  );
+
+  return survey ? <SurveyDisplay survey={survey} /> : <></>;
+}
+
+function SurveyDisplay({ survey }: { survey: Survey }) {
+  const details = React.useRef<HTMLDetailsElement | null>(null);
+
+  const [state, setState] = React.useState(() => getSurveyState(survey.key));
+
+  React.useEffect(() => {
+    writeSurveyState(survey.key, state);
+  }, [state, survey.key]);
+
+  function dismiss() {
+    setState({
+      ...state,
+      dismissed_at: Date.now(),
+    });
+  }
+
+  React.useEffect(() => {
+    const { current } = details;
+    if (!(current instanceof HTMLDetailsElement)) {
+      return;
+    }
+
+    const listener = () => {
+      if (current.open && !state.opened_at) {
+        setState({
+          ...state,
+          opened_at: Date.now(),
+        });
+      }
+    };
+
+    current.addEventListener("toggle", listener);
+
+    return () => current.removeEventListener("toggle", listener);
+  }, [details, state, survey]);
+
+  React.useEffect(() => {
+    if (!state.seen_at) {
+      setState({
+        ...state,
+        seen_at: Date.now(),
+      });
+    }
+  }, [state]);
+
+  if (state.dismissed_at) {
+    return <></>;
+  }
+
+  return (
+    <div className="notecard document-survey">
```

Comment:
Maybe we want something other than a note card? Just so it doesn't look like regular content.

Reply:
@schalkneethling Would you have a good idea of how we could style the survey and what icon we could use?

Reply:
Schalk has shared some ideas on Slack, and we might implement them in a separate PR.